hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -95,6 +95,9 @@
     private final MsgLogger msgLog = MsgLogger.LOGGER;
     private final Logger log = Logger.getLogger(CassDefinitionsServiceImpl.class);
 
+    private Set<DefinitionsEvent> deferredNotifications = new HashSet<>();
```

Comment:
If you agree with the above, I could certainly add some inline comments to make this more clear in the code.